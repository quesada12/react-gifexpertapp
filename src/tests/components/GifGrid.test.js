import { shallow } from "enzyme"
import { GifGrid } from "../../components/GifGrid"
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe('Pruebas en el Componente GifGrid', () => {

    const category = 'Naruto';
    
    test('Debe mostrarse correctamente', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GifGrid category={ category } />);
        expect(wrapper).toMatchSnapshot();
    });
    
    test('Debe de mostrar items cuando se cargan imagenes, useFetchGifs', () => {
        
        const imgs = [{
            id:'ABC',
            url:'https://localhost/cualquiercosa.jpg',
            title:'Cualquier cosa'
        }]

        useFetchGifs.mockReturnValue({
            data: imgs,
            loading: false
        });
        
        const wrapper = shallow(<GifGrid category={ category } />);
        
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(imgs.length);
        
    })
    
    
})
