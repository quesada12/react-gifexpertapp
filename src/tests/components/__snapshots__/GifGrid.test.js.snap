// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el Componente GifGrid Debe de mostrar items cuando se cargan imagenes, useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Naruto
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="https://localhost/cualquiercosa.jpg"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en el Componente GifGrid Debe mostrarse correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Naruto
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
